### ISSUE TITLE
Issues with releases and assets

### ISSUE THREAD

**Issue by hemberger** [MechanicalSoup/MechanicalSoup#239]
1. PyPI does not allow links to local assets in its documentation, so the logo shows up as a missing image.
2. ~~The inclusion of assets in our release file has doubled its size (from 25K to 50K).~~ (edit: actually, no)

From other repositories I have inspected, the README typically has a link to the content on github rather than a local path.

@moy How do you feel about excluding `assets` from the release and linking to our assets on github instead? This probably deserves a post1 release.

**Comment by moy** [comment]
You're right that the logo doesn't show up on PyPI (point 1): there's a `<img alt="MechanicalSoup. A Python library for automating website interaction." src="https://warehouse-camo.cmh1.psfhosted.org/4e918f3614d3b3c95524d2d452f742c1b87f0874/2f6173736574732f6d656368616e6963616c2d736f75702d6c6f676f2e706e67">` link on https://pypi.org/project/MechanicalSoup/, but it's broken.

However, for point 2: I don't see the PNG file in the release (neither by running `setup.py` locally nor in https://pypi.org/project/MechanicalSoup/#files). Did I miss anything?

@senabIsShort: fixing the broken image link this seems feasible for your project to me. The description on PyPI is generated in `setup.py`. We already turn relative links into absolute ones there, so it's just a matter of doing the same on image links.

**Comment by hemberger** [comment]
My mistake -- it was the inclusion of `docs/` that doubled the size of the PyPI tarball, not `assets/` (see #217). So I think we can ignore that point.

**Comment by moy** [comment]
OK, so just point 1 remain valid (I've edited your post to avoid confusion).

**Comment by senabIsShort** [comment]
Sure does look feasible, I'll make this the next item on my bucket list !

**Comment by moy** [comment]
BTW, you can test that the substitution (relative -> absolute link) worked locally with:

```
python3 ./setup.py sdist --keep-temp
head MechanicalSoup-1.0.0.dev0/README.rst
```

A real test on test on https://test.pypi.org/ is welcome too though.

**Comment by QuentinSanchez** [comment]
I've made this code in order to locate the text after "image::" in the readme.
And then concatenate the absolute link to the relative one already there.


```
reg  =r'image::(.*)'
relativeLink= re.search(reg, readme)
relativeLink = relativeLink.group(0)

absoluteLink=relativeLink.replace("whitespace","whitespace 
https://raw.githubusercontent.com/MechanicalSoup/MechanicalSoup/master")
res = re.sub(reg, absoluteLink, readme)
```
 "whitespace" are real whitespace in the code, but i have issues including them in this git message( they create line break).



So i end up with this line in the readme : 

..image:: https://raw.githubusercontent.com/MechanicalSoup/MechanicalSoup/master/assets/mechanical-soup-logo.png

I'am currently trying to set up the test on test.pypi.

**Comment by moy** [comment]
Why would this be more than a `readme = re.sub(..., ..., readme)`, as is done for other substitutions?

If you're starting to have code to show, then best is to submit a pull-request and continue the discussion there.

**Comment by hemberger** [comment]
Has this been fixed by #277?

**Comment by moy** [comment]
Indeed: the "fixes #239" was in the pr description, not the commit message, so the issue wasn't closed.
